prusti-specs/src/export_spec_rewriter.rs

```diff
@@ -0,0 +1,334 @@
+use proc_macro2::{TokenStream, TokenTree};
+use quote::{quote, quote_spanned, ToTokens};
+use syn::spanned::Spanned;
+
+// generate a hash of path
+fn hash_path(path: syn::Path) -> u64 {
+    use std::collections::hash_map::DefaultHasher;
+    use std::hash::{Hash, Hasher};
+    let mut hasher = DefaultHasher::new();
+    path.hash(&mut hasher);
+    hasher.finish()
+}
+
+// generate a macro identifier with given path in this format:
+// {last_segment in path}_{hash(path)}
+pub fn generate_macro_ident(path: syn::Path) -> syn::Ident {
+    let ident = path.segments.last().unwrap().ident.to_owned();
+    syn::Ident::new(&format!("{}_{}_{}", "prusti_export", ident, hash_path(path)), ident.span())
+}
+
+// generate macro for export based on given identifier and item
+fn generate_macro_for_export<T: ToTokens>(
+    macro_ident: syn::Ident,
+    macro_content: T,
+) -> TokenStream {
+    quote_spanned! {macro_ident.span() =>
+        #[macro_export]
+        #[prusti::export_spec_macro]
+        macro_rules! #macro_ident {
+            () => {
+                #macro_content
+            };
+        }
+    }
+}
+
+// helper macro for getting the value of attribute with given name in an item
+// attribute must be in this format: #[name(value)]
+// if attribute is not present then path is not changed
+// @param: $path_mut_ref must be a mutable reference of Path, i.e. &mut syn::Path / mut syn::Path
+macro_rules! get_prusti_path_in_attr {
+    ($item: ident, $attr_str: literal, $path_mut_ref: ident) => {
+        if let Some(export_path_attr) = $item.attrs.iter().find(|attr| {
+            attr.path.segments.len() == 2
+                && &attr.path.segments.last().unwrap().ident.to_string() == $attr_str
+        }) {
+            for tt in export_path_attr.tokens.to_owned() {
+                if let TokenTree::Group(group) = tt {
+                    $path_mut_ref = syn::parse2(group.stream()).unwrap();
+                }
+            }
+        }
+    };
+}
+
+// expand the module path recorded when traversing AST
+fn expand_mod_path_in_macro(
+    path: &syn::Path,
+    mut tokens: TokenStream,
+    uses_same_level: &TokenStream,
+) -> TokenStream {
+    match &path.segments.empty_or_trailing() {
+        true => tokens,
+        false => {
+            for seg in path.segments.to_owned().into_iter().rev() {
```

Comment:
I don't understand why there are multiple places which seem to depend on `empty_or_trailing`… Are you sure you didn't want to just check empty or similar? Is the behaviour actually different if the path is empty here?